packages/components/src/Card/Card.tsx

```diff
@@ -1,26 +1,57 @@
-import React, { ReactNode } from "react";
+import React, { ReactElement, ReactNode } from "react";
 import classnames from "classnames";
 import { XOR } from "ts-xor";
 import styles from "./Card.css";
 import colors from "./colors.css";
 import { CardClickable } from "./CardClickable";
+import { Button, ButtonProps } from "../Button";
 import { Heading } from "../Heading";
 
-interface CardProps {
+interface CardCommonProps {
   /**
    * The `accent`, if provided, will effect the color accent at the top of
    * the card.
    */
   readonly accent?: keyof typeof colors;
   readonly children: ReactNode | ReactNode[];
+}
 
+interface HeaderActionProps {
   /**
    * The title of the card.
    */
   readonly title?: string;
+  /**
+   * The action props that renders into a button on the card header.
+   */
+  readonly action?: Omit<ButtonProps, "size" | "fullWidth">;
```

Comment:
I think we should change this to be a component instead of always a `<Button />`. Since (based on current design vision and usage) the action is either a `<Button />` or `<Menu />`.

![Colin_Blue_-_Jobber](https://user-images.githubusercontent.com/15986172/219385173-cb9f4df9-cd86-464e-a864-df8ec4e1aace.png)

@chris-at-jobber what do you think?

Also prevents us, Atlantis, from having to deprecate this, do a breaking change, update all usage in Jobber Online, mass QA, and release coordination. Makes it easy to extend it too if we do decide to support others like Switch. #futureProofing

---

As for how, you can do it like [InternalChipAffix](https://github.com/GetJobber/atlantis/blob/master/packages/components/src/Chips/InternalChipAffix.tsx)

https://github.com/GetJobber/atlantis/blob/f6dbeb11a162ad2a88556a116b07e4b3d974c972/packages/components/src/Chips/InternalChipAffix.tsx#L11

Setting different defaults can be done in this area
https://github.com/GetJobber/atlantis/blob/f6dbeb11a162ad2a88556a116b07e4b3d974c972/packages/components/src/Chips/InternalChipAffix.tsx#L21-L31

Reply:
That does make sense to me - my concern is Menu right now is pretty opinionated about [it's default activator element](https://github.com/GetJobber/atlantis/blob/4fade5ab2a69d2e13bd645f0a0189ca257b61704/packages/components/src/Menu/Menu.tsx#L96-L105), so for users to easily to match the style they'd need to pass in Menu and a custom activator that defaults to `small` `tertiary` Button.

Could it be simplified so that if > 1 action is passed, the consumer has to provide a menu label but we take care of the "custom" activator and menu options internally? (Or, can the implementation right now be done in a way that allows for us to layer that in easily?)

Reply:
Nice!

I have some side thoughts about making Menu a bit more flexible but we can talk about those another time 😄

Reply:
ah good thing is, it won't change the implementation here so we're safe and should be talked about in another time and place haha

Reply:
updated the action to support `Button` and `Menu`, keeping the same restriction on `Button`. Was chatting with @darryltec for `Menu`, i am not planning to implement any specific restriction on the props. Also neither are not included in the documentation, I suppose to encourage using the `header` directly with title or custom component.